src/QsCompiler/Compiler/Templates/QirDriverCpp.tt

```diff
@@ -1,16 +1,278 @@
 ﻿<#@ template language="C#" linePragmas="false" #>
+<#@ import namespace="Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint" #>
 //------------------------------------------------------------------------------
 // This code was generated by a tool.
 // <auto-generated />
 //------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
 
-// N.B. This is currently just a base non-functional example of a template used
-//      to generate C++ code that works as an entry-point driver for a QIR-based
-//      executable.
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <vector>
 
-extern "C" void <#= entryPointOperation.Name #>(); // NOLINT
+#include "CLI11.hpp"
 
+#include "QirContext.hpp"
+#include "QirRuntime.hpp"
+#include "SimFactory.hpp"
+
+using namespace Microsoft::Quantum;
+using namespace std;
+    
+<# if (entryPointOperation.ContainsArgumentType(DataType.ArrayType)) { #>
+struct InteropArray
+{
+    int64_t Size;
+    void* Data;
+
+    InteropArray(int64_t size, void* data) :
+        Size(size),
+        Data(data){}
+};
+
+template<typename T>
+unique_ptr<InteropArray> CreateInteropArray(vector<T>& v)
+{
+    unique_ptr<InteropArray> array(new InteropArray(v.size(), v.data()));
+    return array;
+}
+
+template<typename S, typename D>
+void TranslateVector(vector<S>& sourceVector, vector<D>& destinationVector, function<D(S&)> translationFunction)
+{
+    destinationVector.resize(sourceVector.size());
+    transform(sourceVector.begin(), sourceVector.end(), destinationVector.begin(), translationFunction);
+
+<# } #>
+<# if (entryPointOperation.ContainsArgumentType(DataType.RangeType) || entryPointOperation.ContainsArrayType(DataType.RangeType)) { #>
```

Comment:
For blocks of code that are conditional, adding a space at the beginning rather than at the end would result in a consistent separation in the generated C++ code.

Reply:
done